### PR TITLE
feat(gateway): implement complete BatteryManagementService

### LINKED ISSUE #183: BatteryManagementService follow-up (Service 3)
Follow-up work to implement Service 3 (BatteryManagementService) once hardware dependency in #158 is ready.

Scope:
- Implement BatteryManagementService RPCs listed in PHASE2_TRACKER.md
- Add HARQ/dispatcher integration and validation
- Add unit tests (15+ cases, 80%+ coverage)
- Register service in cmd/star-gateway/main.go

Notes:
- This is a TODO/follow-up item for the Phase 2 gateway services effort (#181).

### PR DESCRIPTION
**Pull request by CesMag**
## Summary

- Implements all 10 RPC methods for BQ7850 battery management system with comprehensive safety validation
- Adds thread-safe battery state caching with background goroutine and defensive copy pattern
- Implements shutdown registry pattern for deterministic service cleanup in LIFO order
- Achieves 80.38% test coverage with 22 comprehensive tests including edge cases and race detection

## Implementation Details

**BatteryService (654 lines)**
- Request/response methods: GetBatteryState, Get/SetProtectionThresholds, Enable/DisableCellBalancing, GetBalancingStatus, ControlFets, GetDeviceInfo, ResetDevice
- Streaming method: StreamBatteryState with configurable rate (1-10 Hz)
- Li-ion safety validation for voltage (OV/UV), current (OC/OD), temperature (OT/UT)
- Cell balancing bitmask validation for BQ7850 16-cell support
- Constructor nil checks with panic for dependency validation

**Testing (1060 lines, 22 tests)**
- Coverage: 80.38% (exceeds 80% target)
- Race detection: clean
- Edge cases: nil requests, HARQ failures, invalid parameters, zero cell mask, concurrent streaming
- Stream timing constants extracted (no magic numbers)

**Protocol Integration**
- Added BatteryState to wire.proto (field 21)
- Registered MessageTypeBatteryData in dispatcher
- Thread-safe battery state holder with defensive copy via proto.Clone

**Service Architecture**
- Implemented Shutdownable interface for graceful shutdown
- Created shutdown registry pattern in main.go with LIFO shutdown order
- Registered BatteryManagementService with gRPC server
- All safety-critical operations logged at Warn level

## Test Plan

- [x] All 103 service tests pass
- [x] Race detection clean (go test -race)
- [x] Battery service coverage: 80.38%
- [x] Overall coverage: 79.0%
- [x] All validation edge cases tested (nil, invalid, out-of-range)
- [x] StreamBatteryState tested with valid/invalid rates and context cancellation
- [x] Shutdown registry tested with multiple services
- [ ] Manual integration test with grpcurl (requires gateway + firmware running)
- [ ] Hardware test with actual BQ7850 BMS (post-merge)

## Code Quality

- Follows NASA Power of 10 rules (no magic numbers, all constants named)
- Follows SOLID principles (dependency inversion, defensive copies)
- Consistent error handling with gRPC status codes
- All protobuf field names match generated code
- Safety-critical operations logged appropriately

Closes #183

## Summary by CodeRabbit

* **New Features**
  * Battery Management Service: query/stream battery state, configure protection thresholds, control cell balancing and FETs, reset device, retrieve device info, and graceful shutdown integration.

* **Documentation**
  * Expanded implementation and verification docs; added two code-review guidance blocks.

* **Tests**
  * Comprehensive battery service test suite with extensive scenarios and >80% coverage.

* **Chores**
  * Protocol extended to include BatteryState payload; dispatcher and service lifecycle/shutdown updated.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by CesMag** [comment]
Addressed CodeRabbit review comments:
1. Updated `CLAUDE.md` with CodeRabbit introduction and softer workflow language.
2. Updated `BATTERY_SERVICE_PROGRESS.md` with accurate line and test counts (22 tests, ~1070 lines).
3. Added nil checks for `wireMsg` in `battery.go` to prevent potential panics.
4. Fixed potential deadlock in `StreamBatteryState` by using cancellable context and ensuring proper cleanup order.

**Comment by CesMag** [comment]
@coderabbitai full review

**Comment by CesMag** [comment]
@coderabbitai resolve

File: star-gateway/internal/service/battery.go [hunk 1]
```diff
@@ -1,44 +1,656 @@
...
+	minUndervoltageMv  = 2500  // 2.50V
+	maxUndervoltageMv  = 3000  // 3.00V
+	minOverchargeMa    = 1000  // 1A
+	maxOverchargeMa    = 10000 // 10A
+	minOverdischargeMa = 1000  // 1A
+	maxOverdischargeMa = 20000 // 20A
+	minOvertempDeciC   = 450   // 45.0°C
+	maxOvertempDeciC   = 700   // 70.0°C
+	minUndertempDeciC  = -200  // -20.0°C
+	maxUndertempDeciC  = 0     // 0.0°C
```
Comment (bsikar) [inline star-gateway/internal/service/battery.go:43]: We need to talk about these values, I don't think they are correct
